desktop_version/src/Map.cpp

```diff
@@ -396,6 +398,59 @@ void mapclass::transformname(int t)
     }
 }
 
+void mapclass::updateroomnames(void)
+{
+    if (roomnameset)
+    {
+        return;
+    }
+
+    const int rx = game.roomx - 100;
+    const int ry = game.roomy - 100;
+
+    for (int i = specialroomnames.size() - 1; i >= 0; i--)
+    {
+        Roomname &roomname = specialroomnames[i];
+        if (rx == roomname.x && ry == roomname.y && (roomname.flag == -1 || (INBOUNDS_ARR(roomname.flag, obj.flags) && obj.flags[roomname.flag])))
+        {
+            roomname_special = true;
+            if (roomname.type == STATIC)
+            {
+                setroomname(roomname.text[0].c_str());
+            }
+            if (roomname.type == GLITCH)
+            {
+                roomname.delay--;
+                if (roomname.delay <= 0)
+                {
+                    roomname.progress = (roomname.progress + 1) % 2;
+                    roomname.delay = 5;
+                    if (roomname.progress == 0)
+                    {
+                        roomname.delay = 25 + (int) (fRandom() * 10);
+                    }
+                }
+                setroomname(roomname.text[roomname.progress].c_str());
+            }
+            if (roomname.type == TRANSFORM)
+            {
+                roomname.delay--;
+                if (roomname.delay <= 0)
+                {
+                    roomname.progress++;
+                    roomname.delay = 2;
+                    if (roomname.progress >= roomname.text.size())
```

Comment:
This line produces this warning:
```c++
/A/p/r/VVVVVV/desktop_version/src/Map.cpp: In member function ‘void mapclass::updateroomnames()’:
/A/p/r/VVVVVV/desktop_version/src/Map.cpp:442:43: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<std::__cxx11::basic_string<char> >::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
  442 |                     if (roomname.progress >= roomname.text.size())
      |                         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
```
```suggestion
                    if ((size_t) roomname.progress >= roomname.text.size())
```